purldb-toolkit/src/purldb_toolkit/purlcli.py

```diff
@@ -623,64 +703,86 @@ def get_versions(purls, output, file):
     context = click.get_current_context()
     command_name = context.command.name
 
-    purl_versions = list_versions(purls, output, file, command_name)
+    purl_versions = get_versions_details(purls, output, file, unique, command_name)
     json.dump(purl_versions, output, indent=4)
 
 
-# construct_headers() has not yet been implemented for this `versions` command
-# -- or for the `validate` command.
-def list_versions(purls, output, file, command_name):
+def get_versions_details(purls, output, file, unique, command_name):
     """
     Return a list of dictionaries containing version-related data for each PURL
     in the `purls` input list.  `check_versions_purl()` will print an error
     message to the console (also displayed in the JSON output) when necessary.
     """
-    purl_versions = []
-    for purl in purls:
-        purl_data = {}
-        purl_data["purl"] = purl
-        purl_data["versions"] = []
+    versions_details = {}
+    versions_details["headers"] = []
+    versions_details["packages"] = []
+
+    versions_warnings = {}
+
+    input_purls, normalized_purls = normalize_purls(purls, unique)
 
+    clear_log_file()
+
+    for purl in input_purls:
         purl = purl.strip()
         if not purl:
             continue
 
-        versions_purl = check_versions_purl(purl)
-
-        if command_name == "versions" and versions_purl == "not_valid":
-            print(f"'{purl}' not valid")
-            continue
+        purl_data = {}
+        purl_data["purl"] = purl
+        purl_data["versions"] = []
 
-        if command_name == "versions" and versions_purl == "valid_but_not_supported":
-            print(f"'{purl}' not supported with `versions` command")
-            continue
+        versions_purl = check_versions_purl(purl)
 
-        if command_name == "versions" and versions_purl == "not_in_upstream_repo":
-            print(f"'{purl}' does not exist in the upstream repo")
+        if command_name == "versions" and versions_purl:
+            versions_warnings[purl] = versions_purl
             continue
 
-        for package_version_object in list(versions(purl)):
+        for package_version in list(versions(purl)):
             purl_version_data = {}
-            purl_version = package_version_object.to_dict()["value"]
-            nested_purl = purl + "@" + f"{purl_version}"
+            purl_version = package_version.to_dict()["value"]
```

Comment:
I think you can do this:

```suggestion
            purl_version = package_version.value
```